src/Interpreters/RewriteSumIfFunctonVisitor.cpp

```diff
@@ -0,0 +1,104 @@
+#include <AggregateFunctions/AggregateFunctionFactory.h>
+#include <Interpreters/RewriteSumIfFunctonVisitor.h>
+#include <Parsers/ASTFunction.h>
+#include <Parsers/ASTLiteral.h>
+#include <Common/typeid_cast.h>
+
+namespace DB
+{
+
+void RewriteSumIfFunctionMatcher::visit(ASTPtr & ast, Data & data)
+{
+    if (auto * func = ast->as<ASTFunction>())
+        visit(*func, ast, data);
+}
+
+void RewriteSumIfFunctionMatcher::visit(const ASTFunction & func, ASTPtr & ast, Data &)
+{
+    if (!func.arguments || func.arguments->children.empty())
+        return;
+
+    auto lower_name = Poco::toLower(func.name);
+
+    if (lower_name != "sum" && lower_name != "sumif")
+        return;
+
+    auto & func_arguments = func.arguments->children;
+
+    if (lower_name == "sumif")
+    {
+        /// sumIf(1, cond) -> countIf(cond)
+        const auto * literal = func_arguments[0]->as<ASTLiteral>();
+        if (func_arguments.size() == 2 && literal && literal->value.get<Int64>() == 1)
```

Comment:
Probably type of constant should be UInt64, and better to call `tryGet` or check type explicitly. Here and in code below